File: src/main/java/com/battletrident/BattleTrident.java
package com.battletrident;

import com.battletrident.consts.Managers;
import com.battletrident.consts.Plugin;
import com.battletrident.commands.CommandManager;
import com.battletrident.games.player.PlayerManager;
import com.battletrident.games.ring.RingManager;
import com.battletrident.listeners.EventManager;
import com.battletrident.schedulers.ScheduleManager;
import org.bukkit.plugin.java.JavaPlugin;

public class BattleTrident extends JavaPlugin {
	@Override
	public void onEnable() {
		Plugin.Plugin = this;
		Plugin.Logger = this.getLogger();

		Managers.commandManager = new CommandManager();
		Managers.ringManager = new RingManager();
		Managers.scheduleManager = new ScheduleManager();
		Managers.eventManager = new EventManager();
		Managers.playerManager = new PlayerManager();
	}

	@Override
	public void onDisable() {

	}
}

File: src/main/kotlin/com/BattleTrident/BattleTrident.kt
package com.BattleTrident

import com.BattleTrident.Managers.commandManager
import com.BattleTrident.Managers.eventManager
import com.BattleTrident.Managers.playerManager
import com.BattleTrident.Managers.ringManager
import com.BattleTrident.Managers.scheduleManager
import com.BattleTrident.commands.CommandManager
import com.BattleTrident.games.player.PlayerManager
import com.BattleTrident.games.ring.RingManager
import com.BattleTrident.listeners.EventManager
import com.BattleTrident.schedulers.ScheduleManager
import org.bukkit.plugin.java.JavaPlugin

class BattleTrident : JavaPlugin() {
	companion object {
		lateinit var plugin: BattleTrident
	}

	override fun onEnable() {
		plugin = this
		Utils.register(this)

		commandManager = CommandManager()
		playerManager = PlayerManager()
		scheduleManager = ScheduleManager()
		eventManager = EventManager()
		ringManager = RingManager()
	}

	override fun onDisable() {

	}
}